source/audio/appsVolume.py

```diff
@@ -0,0 +1,157 @@
+		activeCallback = None
+
+
+@dataclass(unsafe_hash=True)
+class VolumeSetter(AudioSessionCallback):
+	volumeAndMute: VolumeAndMute | None = None
+
+	def getOriginalVolumeAndMute(self, pid):
+		try:
+			with appVolumesCacheLock:
+				originalVolumeAndMute = appVolumesCache[pid]
+		except KeyError:
+			originalVolumeAndMute = VolumeAndMute(volume=1.0, mute=False)
+		return originalVolumeAndMute
+
+	def onSessionUpdate(self, session: AudioSession) -> None:
+		pid = session.ProcessId
+		simpleVolume = session.SimpleAudioVolume
+		with appVolumesCacheLock:
+			if pid not in appVolumesCache:
+				appVolumesCache[pid] = VolumeAndMute(
+					volume=simpleVolume.GetMasterVolume(),
+					mute=simpleVolume.GetMute(),
+				)
+		if pid != globalVars.appPid:
+			simpleVolume.SetMasterVolume(self.volumeAndMute.volume, None)
+			simpleVolume.SetMute(self.volumeAndMute.mute, None)
+
+	def onSessionTerminated(self, session: AudioSession) -> None:
+		pid = session.ProcessId
+		simpleVolume = session.SimpleAudioVolume
+		originalVolumeAndMute = self.getOriginalVolumeAndMute(pid)
+		try:
+			simpleVolume.SetMasterVolume(originalVolumeAndMute.volume, None)
+			simpleVolume.SetMute(originalVolumeAndMute.mute, None)
+		except Exception:
+			log.exception(f"Could not restore master volume of process {pid} upon exit.")
+
+
+def updateAppsVolumeImpl(
+		volume: float,
+		state: AppsVolumeAdjusterFlag,
+		runTerminators: bool = True,
+):
+	global activeCallback
+	if activeCallback is not None:
+		activeCallback.unregister(runTerminators=runTerminators)
+		activeCallback = None
+	if state == AppsVolumeAdjusterFlag.DISABLED:
+		activeCallback = DummyAudioSessionCallback()
+	else:
+		activeCallback = VolumeSetter(
+			volumeAndMute=VolumeAndMute(
+				volume=volume,
+				mute=state == AppsVolumeAdjusterFlag.MUTED,
+			)
+		)
+	activeCallback.register()
+
+
+def adjustAppsVolume(
+		volumeAdjustment: int | None = None,
+):
+	if not nvwave.usingWasapiWavePlayer():
+		message = _(
+			# Translators: error message when wasapi is turned off.
+			"Other applications` volume cannot be adjusted. "
+			"Please enable WASAPI in the Advanced category in NVDA Settings to use it."
+		)
+		ui.message(message)
+		return
+	volume: int = config.conf["audio"]["applicationsSoundVolume"]
+	state = config.conf["audio"]["ApplicationsVolumeMode"]
+	if state != AppsVolumeAdjusterFlag.ENABLED:
+		# Translators: error message when applications' volume is disabled
+		msg = _("Please enable applications' volume adjuster in order to adjust applications' volume")
+		ui.message(msg)
+		return
+	volume += volumeAdjustment
+	volume = max(0, min(100, volume))
+	config.conf["audio"]["applicationsSoundVolume"] = volume
+
+	# We skip running terminators here to avoid application volume spiking to 100% for a split second.
+	updateAppsVolumeImpl(volume / 100.0, state, runTerminators=False)
+	# Translators: Announcing new applications' volume mesasge
+	msg = _("Applications volume %s") % volume

```

Comment:
Could you use modern string formatting here?
```suggestion
	msg = _("Applications volume {}").format(volume)
```